python/taichi/misc/gui.py

```diff
@@ -235,6 +240,40 @@ def circles(self, pos, color=0xFFFFFF, radius=1):
             raise ValueError(
                 'Color must be an ndarray or int (e.g., 0x956333)')
 
+        if palette is not None:
+            assert palette_indices is not None, 'palette must be used together with palette_indices'
+
+            from taichi.lang.expr import Expr
+
+            if isinstance(palette_indices, Expr):
+                ind_int = palette_indices.to_numpy().astype(np.uint32)
+            elif isinstance(palette_indices, list) or isinstance(
+                    palette_indices, np.ndarray):
+                ind_int = np.array(palette_indices).astype(np.uint32)
+            else:
+                try:
+                    ind_int = np.array(palette_indices)
+                except:
+                    raise TypeError(
+                        'palette_indices must be a type that can be converted to numpy.ndarray'
+                    )
+
+            assert issubclass(
```

Comment:
nit: Note that it's better to raise exceptions in these places. `assert` are meant to verify all kinds of preconditions. When it fails, it detects a bug in Taichi. On the other hand, exceptions are used to detect user input errors :-)